python/pyspark/ml/wrapper.py

```diff
@@ -59,6 +63,32 @@ def _new_java_obj(java_class, *args):
         java_args = [_py2java(sc, arg) for arg in args]
         return java_obj(*java_args)
 
+    @staticmethod
+    def _new_java_primitive_array(pylist):
+        if not pylist:
+            raise ValueError("Unable to convert an empty list to Java array")
```

Comment:
So I understand we can't look at the types on an empty list, but using this seems like a potential headache that might not be worth it - in at least some of the places where we take an array we will want to take an empty array.

Also I forsee this possibly leading to all kinds of fun problems with numeric types :(

What about if we kept `_new_java_array` and just required in our code we specify the type? It should still be simpler when porting new algorithms to Python but give us more flexibility.